src/components/LayoutToggle.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <a v-on:click="toggleLayout()" class="navbar-item is-inline-block-mobile">
+    <span>
+      <i :class="['fas', 'fa-fw', vlayout ? icon : secondaryIcon]"></i>
+    </span>
+    <slot></slot>
+  </a>
+</template>
+
+<script>
+export default {
+  name: "LayoutToggle",
+  props: {
+    name: String,
+    icon: String,
+    iconAlt: String,
+    vlayout: Boolean,
```

Comment:
Why not keeping the generic component and add a `default` prop? I think you could achieve the same but I might have missed something. Idealy reusable components are better and I was thinking of using the SettingsToggle component for a couple of other options I might add at some point.